src/app/locale-selector/locale-selector.module.ts

```diff
@@ -0,0 +1,24 @@
+import { ModuleWithProviders, NgModule } from "@angular/core";
+import { LocalesService, LOCALES_LIST } from "./locales.service";
+import { LocaleSelectorComponent } from "./locale-selector.component";
+import { DropdownSingleSelectModule } from "@backbase/ui-ang/dropdown-single-select";
+import { FormsModule } from "@angular/forms";
+import { CommonModule } from "@angular/common";
+
+@NgModule({
+  declarations: [LocaleSelectorComponent],
+  imports: [DropdownSingleSelectModule, FormsModule, CommonModule],
+  exports: [LocaleSelectorComponent],
+  providers: [ LocalesService ],
+})
+export class LocaleSelectorModule {
+  static forRoot(config: { locales: Array<string> } ): ModuleWithProviders<LocaleSelectorModule> {
```

Comment:
Do we really need that config for a demonstration locale swither?

Reply:
I just wanted to avoid using variables from environment.ts file directly and also wanted to keep that provider inside the locale selector module.